src/components/app/MenuButtons/index.js

```diff
@@ -325,11 +328,17 @@ class MenuButtonsImpl extends React.PureComponent<Props, State> {
   }
 
   _renderPermalink() {
-    const { dataSource, isNewlyPublished, injectedUrlShortener } = this.props;
+    const {
+      dataSource,
+      isNewlyPublished,
+      injectedUrlShortener,
+      profileUrl,
+    } = this.props;
 
     const showPermalink =
       dataSource === 'public' ||
-      dataSource === 'from-url' ||
+      (dataSource === 'from-url' &&
+        !profileUrl.startsWith('http://127.0.0.1')) ||
       dataSource === 'compare';
```

Comment:
What do you think about replacing this line with:
```js
const showPermalink = this.getUploadedStatus(dataSource, profileUrl) === 'uploaded';
```
?

I believe this makes sense, and also we remove this duplication and we can possibly change the logic more easily if necessary in the future (like accepting `https`? accepting more local IPs?)

Reply:
I agree it's better, thanks for the suggestion!